joplin/base/models/janis_page.py

```diff
@@ -175,20 +167,31 @@ def janis_preview_url_end(self, revision=None):
             # Janis will query from its default CMS_API if a param is not provided
             return url_end + f"?CMS_API={settings.CMS_API}"
 
-    # Use hardcoded JANIS_URL for staging and prod
-    # Otherwise use configurable preview_janis_branch setting
-    def janis_url_base(self):
+    def janis_url_base(self, branch):
+        """
+        returns a valid url of the base URL in janis:
+            Use hardcoded JANIS_URL for staging and prod
+            Otherwise, use configurable branch setting
+        """
         if settings.ISSTAGING or settings.ISPRODUCTION:
             return os.getenv("JANIS_URL")
         else:
-            return JanisBranchSettings.objects.first().branch_preview_url_base()
+            branch_settings = JanisBranchSettings.objects.first()
+            return branch_settings.url_base(branch)
```

Comment:
it seems like this line evaluates to:
`return branch_settings.url_base('preview_janis_branch')`

seems a little magic stringy, or maybe the var name branch is throwing me for a loop, not a dealbreaker but something to ponder on

Reply:
I wrote it this way so that you can use either the preview or publish branch, like on line 117 above